sdk/core/Azure.Core/src/Response.cs

```diff
@@ -41,6 +41,39 @@ public abstract class Response : IDisposable
         /// </summary>
         public virtual ResponseHeaders Headers => new ResponseHeaders(this);
 
+        /// <summary>
+        /// Gets the contents of HTTP response, if it is available.
+        /// </summary>
+        /// <remarks>
+        /// Throws <see cref="InvalidOperationException"/> when <see cref="ContentStream"/> is not a <see cref="MemoryStream"/>.
+        /// </remarks>
+        public virtual BinaryData Content
```

Comment:
Do we want to override implementations for existing transports as part of this PR as well?

Reply:
I think the only reason to override it would be if we wanted to cache the binary data, so we didn't have to allocate a new BinaryData object on every call to `.Content`.

I am wondering if in practice that's the right choice to make.  It would cause another field on the `Response` type, to save an allocation in the case where you call `.Content` more than once on a `Response`.  I'm not sure that's worthwhile especially considering that when using the higher level abstractions like `Response<T>` you often don't even touch the response.

If you think it is worthwhile, I am happy to do it (or if there's something better we could do in the concrete implementations that we can't do in the abstraction, let me know) but that is why I did not override implementations for the existing transports.

Reply:
Let's start by not caching.